### ISSUE TITLE
Change resolution / resize

### ISSUE THREAD

**Issue by IBPX** [goatattack/goatattack#28]
I might be missing something, but there doesn't seem to be any way to resize the screen. Even when playing in full screen, it is the same resolution, in the center.

**Comment by freanux** [comment]
The screen size is always a multiple of 640x340. Why that? If you would see more than other players, that could be an advantage. So the screen will be cropped until the next multiplication of 640x340 will fit into the screen.

Hope this helps.

**Comment by IBPX** [comment]
Could there be an option in the settings to manually change the multiple? I would prefer seeing _slightly_ less than having huge black bars around my screen.

Consider possibly having it so that you can run the game at any resolution (you can manually resize), but the scaling will always be the next greatest multiple, so if you tried to make your window bigger so you can see more, you can't. That way you could run it in full screen and still have it equal enough.

Being able to see a little less is not as important as sane scaling, in my opinion.

Imagine the way you can scale the screen in DOOM. Your field of vision doesn't change. You could do something similar with this game, where =/- on the keyboard scale the screen by one multiple. That way, _if you want to_, you can sacrifice less vision for a larger screen.

**EDIT:** [Here's an example of what I'm talking about.](https://youtu.be/mWZPbdYhGuM) (I apologize for the poor quality.)

**Comment by akien-mga** [comment]
An alternative is to implement arbitrary scaling with filtering, the viewport staying always the same but the pixels getting distorted and interpolated. It can typically look a bit worse in pixel art games, but SDL2 likely has facilities for this built-in to make it right.
Then users would have the option to toggle only pixel-perfect scaling (x1, x2, x3, etc.), or arbitrary scaling.

**Comment by IBPX** [comment]
@akien-mga I agree; as much as I dislike interpolated pixel art, I think it would be worth it (gameplay over art for me).
And @freanux, wouldn't it be better to choose a resolution that divides more easily into 1080p/2160p (like 540px tall)? Most displays are 1080p in my experience. It isn't too late to change the native resolution.
